# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLMediaElement.cpp
# Commit: a8a44fb117cb
# Full Hash: a8a44fb117cb6baef601c7ba96b73ac9066d3272
# Author: alwu <alwu@mozilla.com>
# Date: 2020-07-24 09:32:06
# Regressor Bug: 1653390
# File Overlap Count: 1
# Description:
#   Bug 1653390 - part5 : remove 'StopListeningMediaControlKeyIfNeeded()'. r=bryce
#   
#   First, rename `Stop()` to `StopIfNeeded()`.
#   
#   Second, we have already have a check [1] in `StopIfNeeded()` to ensure that that function is only workable when the listener starts, so we don't need `StopListeningMediaControlKeyIfNeeded()` and can call `StopIfNeeded()` directly.
# ==============================================================================

diff -r 3bb773642909 -r a8a44fb117cb dom/html/HTMLMediaElement.cpp
--- a/dom/html/HTMLMediaElement.cpp	Thu Jul 23 21:33:23 2020 +0000
+++ b/dom/html/HTMLMediaElement.cpp	Thu Jul 23 21:33:31 2020 +0000
@@ -426,7 +426,12 @@
     return true;
   }
 
-  void Stop() {
+  /**
+   * Stop listening to the media control keys which would make media not be able
+   * to be controlled via pressing media control keys. If we haven't started
+   * listening to the media control keys, then nothing would happen.
+   */
+  void StopIfNeeded() {
     MOZ_ASSERT(NS_IsMainThread());
     if (!IsStarted()) {
       // We have already been stopped, do not notify stop twice.
@@ -539,7 +544,7 @@
     // `eStart` to the new browsing context. If the media was playing before,
     // we would also notify `ePlayed`.
     bool wasInPlayingState = mState == MediaPlaybackState::ePlayed;
-    Stop();
+    StopIfNeeded();
     Unused << Start();
     if (wasInPlayingState) {
       NotifyMediaStartedPlaying();
@@ -2006,7 +2011,7 @@
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mSeekDOMPromise)
   NS_IMPL_CYCLE_COLLECTION_UNLINK(mSetMediaKeysDOMPromise)
   if (tmp->mMediaControlKeyListener) {
-    tmp->StopListeningMediaControlKeyIfNeeded();
+    tmp->mMediaControlKeyListener->StopIfNeeded();
   }
   NS_IMPL_CYCLE_COLLECTION_UNLINK_WEAK_PTR
 NS_IMPL_CYCLE_COLLECTION_UNLINK_END
@@ -2340,7 +2345,7 @@
   // resume a paused media element.
   ClearResumeDelayedMediaPlaybackAgentIfNeeded();
 
-  StopListeningMediaControlKeyIfNeeded();
+  mMediaControlKeyListener->StopIfNeeded();
 
   // We may have changed mPaused, mAutoplaying, and other
   // things which can affect AddRemoveSelfReference
@@ -4267,7 +4272,7 @@
     mResumeDelayedPlaybackAgent = nullptr;
   }
 
-  StopListeningMediaControlKeyIfNeeded();
+  mMediaControlKeyListener->StopIfNeeded();
   mMediaControlKeyListener = nullptr;
 
   WakeLockRelease();
@@ -6526,7 +6531,7 @@
     mEventDeliveryPaused = true;
     // We won't want to resume media element from the bfcache.
     ClearResumeDelayedMediaPlaybackAgentIfNeeded();
-    StopListeningMediaControlKeyIfNeeded();
+    mMediaControlKeyListener->StopIfNeeded();
   } else {
     if (!mPaused) {
       mCurrentLoadPlayTime.Start();
@@ -7875,12 +7880,6 @@
   }
 }
 
-void HTMLMediaElement::StopListeningMediaControlKeyIfNeeded() {
-  if (mMediaControlKeyListener->IsStarted()) {
-    mMediaControlKeyListener->Stop();
-  }
-}
-
 void HTMLMediaElement::UpdateMediaControlAfterPictureInPictureModeChanged() {
   // Hasn't started to connect with media control, no need to update anything.
   if (!mMediaControlKeyListener->IsStarted()) {